code/datums/components/butchering.dm

```diff
@@ -161,3 +161,47 @@
 		return
 	if(victim.stat == DEAD && (victim.butcher_results || victim.guaranteed_butcher_results))
 		Butcher(parent, victim)
+
+/datum/component/butchering/wearable
+
+/datum/component/butchering/wearable/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_MECHA_DRILL_MOB, .proc/Butcher)
+	RegisterSignal(parent, list(COMSIG_ITEM_EQUIPPED, COMSIG_MECHA_EQUIPMENT_ATTACHED), .proc/enable_butchering)
+	RegisterSignal(parent, list(COMSIG_ITEM_DROPPED, COMSIG_MECHA_EQUIPMENT_DETACHED), .proc/disable_butchering)
+
+/datum/component/butchering/wearable/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(
+		COMSIG_MECHA_DRILL_MOB,
+		COMSIG_ITEM_EQUIPPED,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_MECHA_EQUIPMENT_ATTACHED,
+		COMSIG_MECHA_EQUIPMENT_DETACHED
+		))
+
+///Enables the butchering mechanic for the mob who has equipped us.
+/datum/component/butchering/wearable/proc/enable_butchering(datum/source, mob/user, slot)
+	SIGNAL_HANDLER
+	// check if slot is valid
+	if(slot == ITEM_SLOT_HANDS)
+		return
+	butchering_enabled = TRUE
+	// check wheter this was called by a different signal then COMSIG_ITEM_EQUIPPED
+	if(!slot)
+		return
+	RegisterSignal(user, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, .proc/butcher_target)
+
+///Disables the butchering mechanic for the mob who has equipped us.
+/datum/component/butchering/wearable/proc/disable_butchering(datum/source, mob/user, slot)
+	SIGNAL_HANDLER
+	butchering_enabled = FALSE
+	if(!slot)
+		return
```

Comment:
so I should have explained myself better, item_equipped and item_dropped signal send the slot flag, the mechas dont I used that to determine wheter or not this signal was send by mecha since they dont register the COMSIG_HUMAN_EARLY_UNARMED_ATTACK, and I only hard checked the hands since the hostile_environment suit only sends the item_dropped or item_equipped signals once you toggle the hood, but I should have seperated them for future proofing

Reply:
did it all like you requested